tests/python/unittest/test_tvmscript_ops.py

```diff
@@ -0,0 +1,90 @@
+import tvm
+from tvm.script import ty
+from tvm import te, tir
+import numpy as np
+import tvm.testing
+
+
+@tvm.script.tir
+def get_valid_counts(
+    data: ty.handle,
+    valid_count: ty.handle,
+    out: ty.handle,
+    out_indices: ty.handle,
+    score_threshold: ty.float32,
+    id_index: ty.int32,
+    score_index: ty.int32,
+) -> None:
+
+    data_buf = tir.match_buffer(data, (1, 2500, 6), "float32")
+    valid_count_buf = tir.match_buffer(valid_count, (1,), "int32")
+    out_buf = tir.match_buffer(out, (1, 2500, 6), "float32")
+    out_indices_buf = tir.match_buffer(out_indices, (1, 2500), "int32")
```

Comment:
Thanks. I wonder what if the buffer type is not matched with the parameter? Suggest to also add a bad case test.

Reply:
if the buffer type is not matched with the parameter,  an error will be reported as below When it is bulit,
`Check failed: (dtype.element_of() == n->dtype.element_of() && dtype.lanes() % n->dtype.lanes() == 0) is false: Cannot store float32 to buffer of float16` 

IMO, This error is not only reflected in the buffer type is not matched with the parameter， two buffer type which is not matched, the same problem will also occur. eg (a buffer is float32, out buffer is float16):

```
@tvm.script.tir
def func(a: ty.handle, out:ty.handle) -> None:
    out_buf = tir.match_buffer(out, (1, 5), "float16")
    in_buf = tir.match_buffer(a, (1), "float32")
    with tir.block([1, 5], "init") as [vi, vj]:
        out_buf[vi, vj] = in_buf[0]

np_data = np.zeros(shape=(1, 5)).astype("float16")
score = 0.5
out_data = tvm.nd.array(np_data, ctx)
score_data = tvm.nd.array(np.array([score], dtype="float32"), ctx) 
f(score_data, out_data) 

```

So when we write script functions, we must avoid such errors.